extensions/amp-subscriptions/0.1/platform-store.js

```diff
@@ -365,6 +365,24 @@ export class PlatformStore {
     return entitlementsResolved === this.serviceIds_.length;
   }
 
+  /**
+   * Calculates weight to add/remove based on getSupportedFactor()
+   * @param {string} factorName
+   * @param {!./subscription-platform.SubscriptionPlatform} platform
+   * @return {!Promise<number>}
+   * @private
+   */
+  getSupportedFactorWeight_(factorName, platform) {
+    return platform.getSupportedFactor(factorName)
+        .then(factorValue => {
+          if (!factorValue) {
```

Comment:
when factorValue is a negative number this will fail. For example
! (-1) is equal to false